addons/web_tour/static/src/tour_service/tour_service.js

```diff
@@ -288,7 +289,21 @@ export const tourService = {
             const macro = convertToMacro(tour, pointer, options);
             const willUnload = callWithUnloadCheck(() => {
                 if (tour.url && tour.url !== options.startUrl && options.redirect) {
-                    window.location.href = window.location.origin + tour.url;
+                    const fromUrl = new URL(browser.location.href);
+                    const toUrl = new URL(browser.location.origin + tour.url);
+                    const debugMode = fromUrl.searchParams.get("debug");
+                    if (debugMode) {
+                        const allDebug = [];
+                        [toUrl.searchParams.get("debug") || "", debugMode].forEach((string) => {
+                            string.split(",").forEach((mode) => {
+                                if (mode && !allDebug.includes(mode)) {
+                                    allDebug.push(mode);
+                                }
+                            });
+                        });
+                        toUrl.searchParams.set("debug", sortBy(allDebug).join(","));
+                    }
+                    browser.location.href = toUrl.href;
```

Comment:
Maybe we can use the `redirect` function from urls.js

Reply:
Doesn't exist yet in 16.3